### ISSUE TITLE
Default to ignoring Temp Tables for Compression check

### ISSUE THREAD

**Issue by LowlyDBA** [tsqllint/tsqllint#275]
I believe in most cases it isn't beneficial to mandate compression for temporary tables.

https://github.com/tsqllint/tsqllint/blob/cfa67dadfc7820c5c77ea73a8b622ada8b0e5dc9/source/TSQLLint.Infrastructure/Rules/DataCompressionOptionRule.cs#L24-L33

Requesting that this not apply at all to temp tables, or be configurable with a default of not applying to temp tables.

**Comment by nathan-boyd** [comment]
This will be tricky to handle. Changing the rule would be a breaking change for existing users, creating sub-config for each rule is an option, the schema doesn't currently support it but it wouldn't be too major of a change.

**Comment by LowlyDBA** [comment]
Yeah, the breaking change aspect does make it a bit more nuanced. 

This is probably a low-priority change, but might be useful to integrate for new rules that create/alter tables as they come up. For now we'll probably add guidance on using the exclusion comments for these scenarios.
